client/clienterror/clienterror.go

```diff
@@ -103,6 +103,32 @@ func New(err error) *APIError {
 		}
 	}
 	if createClusterDefaultErr, ok := err.(*clusters.AddClusterDefault); ok {
+		ae := &APIError{
+			HTTPStatusCode: createClusterDefaultErr.Code(),
+			OriginalError:  createClusterDefaultErr,
+			ErrorMessage:   createClusterDefaultErr.Error(),
+		}
+		if ae.HTTPStatusCode == http.StatusNotFound {
+			ae.ErrorMessage = "Not found"
+			ae.ErrorDetails = "A 404 error has occurred when attempting to create the cluster."
+		} else if ae.HTTPStatusCode == http.StatusBadRequest {
+			ae.ErrorMessage = "Invalid parameters"
+			ae.ErrorDetails = "The cluster cannot be created. Some parameter(s) are considered invalid.\n"
+			ae.ErrorDetails += "Details: " + createClusterDefaultErr.Payload.Message
+		}
+		return ae
+	}
+
+	// create cluster V5
+	if createClusterUnauthorizedErr, ok := err.(*clusters.AddClusterV5Unauthorized); ok {
```

Comment:
I was thinking if we can somehow use the new `errors.As()` here to make this cleaner, but my quick test failed. Just dropping a note here in case someone figures directly how it would work. I'm ok with this for now.